### DISCUSSION TITLE
steamer example for which directly read data from RTP server

### DISCUSSION THREAD

**Discussion by manthanrtilva** [paullouisageneau/libdatachannel#1306] [Q&A]
In my application, I need to read audio/video data from from RTP server and send it to browser.

Using media-sender I am able to send video data to browser.

I tried to add new track to peerconnection and send data from rtp socket to newly created track, but browser don't playing sound.

Then I checked streamer example, which is playing audio and video, but I am looking for a way to create new `StreamSource` which read from RTP server and rest can be handle by existing stream example.

Any hint how to do this?

**Comment by paullouisageneau** [comment]
If your stream is already packetized as RTP, you must not proceed like the streamer example but like the media-sender one. Creating a new track to send audio is the correct way. If audio is not playing it means there is a mistake somewhere. You should check the track description information matches the audio RTP flow (Codec, payload type, SSRC, etc), and look at `about:webrtc` or `about:webrtc-internals` for hints of what's wrong with playback.
